sda-commons-server-opentelemetry/src/main/java/org/sdase/commons/server/opentelemetry/OpenTelemetryBundle.java

```diff
@@ -0,0 +1,187 @@
+  private static final String JAEGER_SAMPLER_PARAM = "JAEGER_SAMPLER_PARAM";
+  private final OpenTelemetry openTelemetry;
+  private final Pattern excludedUrlPatterns;
+
+  public OpenTelemetryBundle(OpenTelemetry openTelemetry, Pattern excludedUrlPatterns) {
+    this.openTelemetry = openTelemetry;
+    this.excludedUrlPatterns = excludedUrlPatterns;
+  }
+
+  @Override
+  public void run(Configuration configuration, Environment environment) {
+    registerJaxrsTracer(environment.jersey());
+    registerServletTracer(environment, this.openTelemetry);
+    registerAdminTracer(environment, this.openTelemetry);
+  }
+
+  @Override
+  public void initialize(Bootstrap<?> bootstrap) {
+    // nothing here
+  }
+
+  public void registerServletTracer(Environment environment, OpenTelemetry openTelemetry) {
+    TracingFilter tracingFilter = new TracingFilter(openTelemetry, excludedUrlPatterns);
+
+    FilterRegistration.Dynamic filterRegistration =
+        environment.servlets().addFilter("TracingFilter", tracingFilter);
+    filterRegistration.addMappingForUrlPatterns(
+        EnumSet.of(DispatcherType.REQUEST, DispatcherType.ASYNC), false, "*");
+  }
+
+  public void registerAdminTracer(Environment environment, OpenTelemetry openTelemetry) {
+    TracingFilter adminTracingFilter = new TracingFilter(openTelemetry, excludedUrlPatterns);
+
+    FilterRegistration.Dynamic adminFilterRegistration =
+        environment.admin().addFilter("AdminTracingFilter", adminTracingFilter);
+    adminFilterRegistration.addMappingForUrlPatterns(
+        EnumSet.of(DispatcherType.REQUEST, DispatcherType.ASYNC), false, "*");
+  }
+
+  public void registerJaxrsTracer(JerseyEnvironment jersey) {
+    jersey.register(JerseyExceptionListener.class);
+    jersey.register(new ServerTracingFilter());
+  }
+
+  private static OpenTelemetry bootstrapConfiguredTelemetrySdk() {
+    LOG.info("No OpenTelemetry sdk instance is provided, using autoConfigured instance.");
+    return AutoConfiguredOpenTelemetrySdk.builder()
+        .addPropertiesSupplier(SdaConfigPropertyProvider::getProperties)
+        .setResultAsGlobal(shouldSetAsGlobal())
+        .build()
+        .getOpenTelemetrySdk();
+  }
+
+  public static InitialBuilder builder() {
+    return new Builder();
+  }
+
+  public interface InitialBuilder {
+    /**
+     * Specifies a custom telemetry instance to use. If this builder is used, this module will only
+     * be a consumer, and the provider needs to take care of registering this instance as global.
+     *
+     * @param openTelemetry The telemetry instance to use.
+     * @return the same builder
+     */
+    FinalBuilder withTelemetryInstance(OpenTelemetry openTelemetry);
+
+    /**
+     * Enables the bundle to setup an auto configured instance of OpenTelemetry Sdk, and register it
+     * as Global.
+     *
+     * @return the same builder.
+     */
+    FinalBuilder withAutoConfiguredTelemetryInstance();
```

Comment:
this builder provides two options, you can either provide an OpenTelemetry instance if you already have created one, or use `withAutoConfiguredTelemetryInstance` so that it creates a new instance by itself.